lib/date_range.rb

```diff
@@ -0,0 +1,52 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    
+    attr_accessor :start_date, :end_date
+
+    # your library code should assume that it's receiving Date objects to start
+    def initialize(start_date, end_date)
+      raise ArgumentError.new("Please pass in Date class instances.") if !(start_date.is_a? Date) || !(end_date.is_a? Date)
+      raise ArgumentError.new("Live in the present! Dates should not be prior to today.") if start_date < Date.today
```

Comment:
It's helpful to include the bad arguments in message of the `ArgumentError`:

```suggestion
      raise ArgumentError.new("Live in the present! Dates should not be prior to today.  Bad start_date: #{start_date}") if start_date < Date.today
```